### PR TITLE
refactor: improve training reproducibility and scaler persistence

### LINKED ISSUE #3032: Improve:
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What would you like to improve?

# Improve Training Pipeline Reproducibility and Remove Duplicate Scaler Persistence

## Describe the feature

The training pipeline currently saves the fitted scaler multiple times and does not provide reproducible training runs.

#### 1. Duplicate Scaler Persistence

The scaler is saved inside `scale()` when `fit=True`:

```python
joblib.dump(scaler, SCALER_PATH)
```

and then saved again later in `train()`:

```python
joblib.dump(scaler, SCALER_PATH)
```

This duplicates responsibility for artifact persistence and makes the data preprocessing function perform file-system side effects.

#### 2. Non-Reproducible Training

The simulated training data relies on random generation:

```python
np.random.randint(...)
```

but no random seed is configured.

As a result:

* Training results vary between runs.
* Generated thresholds may differ.
* Hyperparameter tuning outcomes are harder to reproduce.
* Debugging model behavior becomes more difficult.

### Scaler Persistence

* Move scaler saving responsibility to a single location.
* Prefer keeping persistence logic in `train()` and keeping `scale()` focused on preprocessing.

### Reproducibility

* Introduce a configurable random seed (e.g. `RANDOM_SEED = 42`).
* Initialize NumPy's random generator before dataset generation.

## Benefits

* Cleaner separation of concerns.
* Easier maintenance.
* Reproducible experiments and model artifacts.
* More predictable threshold generation.
* Improved debugging and testing experience.

## Additional Context

This change is focused on training pipeline reliability and code quality. No model architecture changes are required.

### Why does this matter?

_No response_

### Screenshots or screen recording (if applicable)

_No response_

### Record

- [x] I agree to follow this project's Code of Conduct
- [x] I want to work on this issue

### PR DESCRIPTION
**Pull request by Ommdutta26**
## Screenshot (when helpful)

N/A – This PR contains training pipeline refactoring and reproducibility improvements. No UI changes were made.

---

## What this PR does

This PR improves the training pipeline by making training runs reproducible and removing duplicate scaler persistence.

### Changes included

* Added a fixed random seed (`RANDOM_SEED = 42`) to ensure reproducible training data generation and model training behavior.
* Removed duplicate scaler saving logic from `scale()`.
* Kept scaler persistence centralized within `train()` to maintain clear separation of responsibilities.
* Added validation for generated session counts to prevent invalid inputs.

### Benefits

* More reproducible experiments and debugging.
* Cleaner separation of concerns.
* Reduced redundant file operations.
* Improved maintainability of the training pipeline.

---

## Type of change

* [x] Bug fix
* [ ] New feature
* [x] Code refactor
* [ ] Documentation update

---

## Related issue

Closes #3032

---

## More screenshots (optional)

N/A

### CODE REVIEW

**Comment by Ommdutta26** [comment]
@ronisarkarexe kindly review this PR its pending from 4 days
